### PR TITLE
Upgrade postcss to v8.x

### LINKED ISSUE #33: PostCSS 8 Support
Looks like this plugin needs to be updated still for PostCSS 8.  Here is the migration guide for anyone willing to do it:

https://evilmartians.com/chronicles/postcss-8-plugin-migration

I only have time right now to subscribe to the issue and leave that breadcrumb though.

### PR DESCRIPTION
**Pull request by kmye**
Fixes https://github.com/postcss/postcss-browser-reporter/issues/33

- Upgrade PostCSS version to v8
- Move postcss from dependencies to devDependencies and peerDependencies
- Update index.js to use the updated API

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,89 +1,96 @@
...
+
+      root.last.append({ prop: 'content', value: `"${content}"` });
     }
-});
+  }
+}
+
+plugin.postcss = true;
+
+module.exports = plugin;
```
Comment (ai) [inline index.js]: Text editor doesn’t follow `.editorconfig` settings

File: index.js [hunk 2]
```diff
@@ -1,89 +1,96 @@
...
-                    selector = 'html::after';
-                }
-            });
-        }
-
-        css.append({ selector: selector });
-        for ( var style in styles ) {
-          if ( styles.hasOwnProperty(style) ) {
-            css.last.append({ prop: style, value: styles[style] });
+const postcss = require('postcss');
```
Comment (ai) [inline index.js]: Why do we need import?

File: index.js [hunk 3]
```diff
@@ -1,89 +1,96 @@
...
+  }
+  return str;
+}
+
+const plugin = (opts = {}) => {
+  if (opts?.disabled === true) {
+    return function () { };
+  }
+
+  const defaultStyles = {
```
Comment (ai) [inline index.js]: We need to export it to root level

File: index.js [hunk 4]
```diff
@@ -1,89 +1,96 @@
...
+    'white-space': 'pre-wrap',
+    'font-family': 'Menlo, Monaco, monospace',
+    'text-shadow': '0 1px #A82734'
+  };
+
+  const styles = (opts?.styles ?? defaultStyles);
+
+  return {
+    postcssPlugin: 'postcss-browser-reporter',
+    Once(root, { result }) {
```
Comment (ai) [inline index.js]:
```suggestion
    OnceExit(root, { result }) {
```

I think we need to call it after all plugins. Otherwise `warnings` will be empty.

**Comment by ai** [comment]
Do you really use this plugin? Releasing it will take an hour from me, I prefer to do it only if somebody will have real practical result from it.

File: index.js (hunk 4 above)
Comment (kmye) [reply index.js]: changing this to OnceExit will result in the subsequent warning being removed from the content

**Comment by kmye** [comment]
> Do you really use this plugin? Releasing it will take an hour from me, I prefer to do it only if somebody will have real practical result from it.

yes, we used in our project as getting vulnerability warning from this plugin because the dependency is an older version and installing this plugin in the project also installs the old postcss which has vulnerability.

if this plugin is not maintained, then we might consider to not use it all at.. what do you think?

File: index.js (hunk 4 above)
Comment (ai) [reply index.js]: > changing this to OnceExit will result in the subsequent warning being removed from the content

What do you mean? Can you show some example?

**Comment by ai** [comment]
> if this plugin is not maintained, then we might consider to not use it all at.. what do you think?

It is not very hard to maintain this simple plugin. I will release changes.

File: index.js (hunk 4 above)
Comment (kmye) [reply index.js]: tried to removed, then the test broke
```
content: "1:99       Warning with details [bar]\00000aHere is another warning"
```

File: index.js (hunk 1 above)
Comment (kmye) [reply index.js]: oops my bad 😅

File: index.js (hunk 4 above)
Comment (ai) [reply index.js]: Test `displays only warnings from plugins before` should be removed for PostCSS 8. New event system removed plugins order idea.

File: index.js (hunk 3 above)
Comment (kmye) [reply index.js]: sorry i'm not very familiar with. what's the use case for exporting out defaultStyles? don't see the usage for this

Comment (ai) [reply index.js]: No need for export just move `const defaultStyles` out of the function to the root level of the file.

It has the same content, so we can define it only once (and not on every function call).

Comment (kmye) [reply index.js]: or do you mean moving out of line 19 to after line 13? or export as in module.exports ? 😅

Comment (ai) [reply index.js]: > or export as in module.exports ?

No

> do you mean moving out of line 19 to after line 13

Yes, right now it is good

File: index.js (hunk 4 above)
Comment (kmye) [reply index.js]: okay, i have updated the to use OnceExit and also updated the tests. 😄

**Comment by ai** [comment]
I updated configs https://github.com/postcss/postcss-browser-reporter/commit/9eda33234e05f216d5df5a2e7e270a0bf59977bd

**Comment by ai** [comment]
Released in the version 0.7.

**Pull request merged**